### PR TITLE
fix: move list_core_wallets() RPC off the UI thread

### PR DESCRIPTION
**Pull request by lklimek**
## Summary

- Move synchronous `list_core_wallets()` RPC calls off the UI thread to prevent freezes when Dash Core is slow or unreachable
- Add `CoreTask::ListCoreWallets` backend task + `BackendTaskSuccessResult::CoreWalletsList` result variant
- Convert all three call sites (AddNewWalletScreen, ImportMnemonicScreen, WalletsBalancesScreen) to async fetch pattern
- Add error recovery (retry on failure) and network-switch cache reset

Ref #700

## Test plan

- [ ] Manual test scenarios: [`docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md`](docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md)
- [ ] Verify Create Wallet screen renders immediately without freeze
- [ ] Verify Import Wallet screen renders immediately without freeze
- [ ] Verify CoreWalletNotConfigured error triggers async wallet list fetch
- [ ] Verify network switch re-fetches Core wallet list

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

## Summary by CodeRabbit

* **New Features**
  * Wallet lists now load in the background for smoother UI responsiveness.
  * Auto-select when exactly one wallet is found; enhanced multi-wallet selection dialog when multiple results exist.
  * Deferred retry/re-fetch on network changes or errors; screens now reset wallet cache/state when switching.

* **Bug Fixes**
  * Improved handling of missing Core wallet data to avoid spurious error banners and recover gracefully.

* **Documentation**
  * Added comprehensive manual test scenarios for async wallet-list flows, error recovery, and edge cases.

### CODE REVIEW

File: src/ui/wallets/add_new_wallet_screen.rs [hunk 1]
```diff
@@ -717,7 +727,25 @@ impl AddNewWalletScreen {
 }
 
 impl ScreenLike for AddNewWalletScreen {
+    fn display_task_result(&mut self, backend_task_success_result: BackendTaskSuccessResult) {
+        if let BackendTaskSuccessResult::CoreWalletsList(wallets) = backend_task_success_result {
+            self.core_wallets = Some(wallets);
+        }
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/ui/wallets/add_new_wallet_screen.rs:736]: ✅ Fixed in 9d4960b6 — `selected_core_wallet_index` is now clamped via `.min(wallets.len().saturating_sub(1))` before storing the refreshed list.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/ui/wallets/import_mnemonic_screen.rs [hunk 2]
```diff
@@ -523,7 +533,25 @@ impl Drop for ImportMnemonicScreen {
 }
 
 impl ScreenLike for ImportMnemonicScreen {
+    fn display_task_result(&mut self, backend_task_success_result: BackendTaskSuccessResult) {
+        if let BackendTaskSuccessResult::CoreWalletsList(wallets) = backend_task_success_result {
+            self.core_wallets = Some(wallets);
+        }
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/ui/wallets/import_mnemonic_screen.rs:542]: ✅ Fixed in 9d4960b6 — same clamp applied here.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+### Steps
+1. Navigate to "Create Wallet"
+2. Complete all required steps
+3. Observe the form before the "Save Wallet" button
+
+### Expected Results
+- Step 3: No "Dash Core Wallet" ComboBox appears at any point during the session
+- Step numbering remains at "6. Save the wallet." (no extra Core wallet step)
+- Clicking "Save Wallet" saves the wallet with `core_wallet_name = NULL` in SQLite
+
```
[the remark this answers is not in the document]
Comment (lklimek) [reply docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md:102]: Not reproduced. The save logic uses `.get(self.selected_core_wallet_index)` which safely returns `None` when OOB. When exactly one wallet exists and the ComboBox is hidden, index 0 correctly auto-selects it — this is intentional UX, not a bug. Corrected the test scenario doc to match actual behavior (auto-assigns sole wallet, `NULL` only when zero wallets loaded) in 9d4960b6.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/backend_task/mod.rs [hunk 4]
```diff
@@ -271,6 +271,9 @@ pub enum BackendTaskSuccessResult {
 
     // Mining results (dev mode, Regtest/Devnet only)
     MineBlocksSuccess(u64),
+
+    // Core wallet list (async fetch of loaded Core wallets)
+    CoreWalletsList(Vec<String>),
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/backend_task/mod.rs:276]: Not reproduced. `change_network()` (app.rs:874) only calls `change_context()` on `main_screens`, NOT on `screen_stack`. These modal screens never receive a new `app_context` on network switch, so adding network metadata to the enum would not help — the entire screen remains stale. This is a pre-existing architectural limitation, not something introduced or fixable in this PR.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/ui/wallets/wallets_screen/mod.rs [hunk 5]
```diff
@@ -2010,6 +2019,12 @@ impl ScreenLike for WalletsBalancesScreen {
             }
         }
 
+        // Dispatch the async ListCoreWallets task if pending
+        if self.pending_list_core_wallets {
+            self.pending_list_core_wallets = false;
+            action |= AppAction::BackendTask(BackendTask::CoreTask(CoreTask::ListCoreWallets));
+        }
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/ui/wallets/wallets_screen/mod.rs:2032]: Not reproduced. Would require two distinct wallet operations to fail with `CoreWalletNotConfigured` simultaneously — practically unreachable. The in-flight guard adds complexity (3 new fields, multiple coordination points) for negligible benefit.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+### Expected Results
+- Step 3: `reset_core_wallets_cache()` is called via `change_context()`, clearing `core_wallets`
+  and resetting `core_wallets_loading` to `false`
+- Step 4: The ComboBox temporarily disappears (no stale Mainnet data is shown)
+- Step 5: A new `ListCoreWallets` task is dispatched for Testnet
+- After the task completes, the ComboBox lists only `testnet_x` (the single Testnet wallet),
+  or is absent entirely per ACW-003 behavior
+- No Mainnet wallet names appear after the switch
+
```
[the remark this answers is not in the document]
Comment (lklimek) [reply docs/ai-design/2026-03-10-async-list-core-wallets/manual-test-scenarios.md]: ✅ Fixed in 204df56d — ACW-012/013 rewritten to reflect that these modal screens are on `screen_stack` and do NOT receive `change_context()` from `change_network()`. Stale data persists until dismissed and reopened.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/ui/wallets/add_new_wallet_screen.rs [hunk 7]
```diff
@@ -717,7 +727,28 @@ impl AddNewWalletScreen {
...
+        false
+    }
+
     fn ui(&mut self, ctx: &Context) -> AppAction {
+        let mut pending_action = AppAction::None;
+        if self.core_wallets.is_none() && !self.core_wallets_loading {
+            self.core_wallets_loading = true;
+            pending_action =
+                AppAction::BackendTask(BackendTask::CoreTask(CoreTask::ListCoreWallets));
+        }
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/ui/wallets/add_new_wallet_screen.rs:751]: ✅ Fixed in 204df56d — `display_task_error()` now sets `self.core_wallets = Some(vec![])` to break the retry loop. Users navigate away and back to retry.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>

File: src/ui/mod.rs [hunk 8]
```diff
@@ -761,14 +761,20 @@ impl Screen {
...
+            Screen::AddNewWalletScreen(screen) => {
+                screen.app_context = app_context;
+                screen.reset_core_wallets_cache();
+            }
             Screen::TransferScreen(screen) => screen.app_context = app_context,
             Screen::TopUpIdentityScreen(screen) => screen.app_context = app_context,
             Screen::WalletsBalancesScreen(screen) => {
                 screen.app_context = app_context;
                 screen.update_selected_wallet_for_network();
             }
```
[the remark this answers is not in the document]
Comment (lklimek) [reply src/ui/mod.rs:774]: ✅ Fixed in 204df56d — added `reset_pending_list_state()` to `WalletsBalancesScreen`, called from `change_context()` before `update_selected_wallet_for_network()`.

<sub>🤖 Co-authored by [Claudius the Magnificent](https://github.com/lklimek/claudius) AI Agent</sub>
